shell/browser/hid/hid_chooser_controller.cc

```diff
@@ -0,0 +1,327 @@
+// Copyright (c) 2021 Microsoft, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "shell/browser/hid/hid_chooser_controller.h"
+
+#include <utility>
+
+#include "base/bind.h"
+#include "base/command_line.h"
+#include "base/containers/contains.h"
```

Comment:
this is copied from https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/hid/hid_chooser_controller.cc, yeah? is there any way we could reuse/call into any of these files from upstream?

Reply:
@codebytere we would have to patch a decent amount because chrome/browser/ui/hid/hid_chooser_controller.cc is a GUI chooser versus Electron firing an event.  Out of curiosity, I created a patch file and its pretty large: [feat_add_support_for_webhid.txt](https://github.com/electron/electron/files/6862670/feat_add_support_for_webhid.txt).  We could trim it down somewhat I guess but it definitely isn't a small change.